### DISCUSSION TITLE
Wait for full deployment before checking contract source code

### DISCUSSION THREAD

**Discussion by SimonR-Nano** [NomicFoundation/hardhat#4650] [General]
Hello! 
I'm trying to verify my contract but i got:
```
Reason: The Etherscan API responded that the address 0x... does not have bytecode.
This can happen if the contract was recently deployed and this fact hasn't propagated to the backend yet.
Try waiting for a minute before verifying your contract. If you are invoking this from a script,
try to wait for five confirmations of your contract deployment transaction before running the verification subtask.
```
Is it possible to wait the full deployement without making a "delay" with a typed value... Like `await delay(10000)`...

**Comment by fvictorio** [comment]
It depends on the library you are using. In ethers you can use `await contract.waitfordeployment()`. With hardhat-viem contract deployments are waited by default, but you can pass an optional `confirmations` option to wait for a specific number of block confirmations.

**Reply by SimonR-Nano** [reply]
Thanks, used: `await contract.deploymentTransaction()?.wait(4);` !
